backend/service/github/github.go

```diff
@@ -621,3 +622,71 @@ func (s *svc) DeleteFile(ctx context.Context, ref *RemoteRef, path, sha, message
 	}
 	return contentRes, nil
 }
+
+func (s *svc) CreateCommit(ctx context.Context, ref *RemoteRef, message string, files FileMap) (*Commit, error) {
```

Comment:
Most of this functionality is identical to the CreateBranch function, so I think we can pull out the similar pieces into a helper function and utilize it in both places so we don't have duplicate logic.

Reply:
Sounds good! I'll wait until we merge the changes in #3106 to encapsulate this logic.